deployments/puppet/manifests/init.pp

```diff
@@ -1,15 +1,52 @@
 # Main class that installs and configures the agent
 class signalfx_agent (
-    $config,
-    $package_stage = 'final',
-    $repo_base = 'dl.signalfx.com',
-    $config_file_path = '/etc/signalfx/agent.yaml',
-    $version = 'latest') {
+  $config,
+  $package_stage    = 'final',
+  $repo_base        = 'dl.signalfx.com',
+  $config_file_path = $::osfamily ? {
+    'debian'  => '/etc/signalfx/agent.yaml',
+    'redhat'  => '/etc/signalfx/agent.yaml',
+    'windows' => 'C:\\ProgramData\\SignalFxAgent\\agent.yaml',
+    'default' => '/etc/signalfx/agent.yaml'
+  },
+  $agent_version    = '',
+  $package_version = 'latest',
+) {
+
+  $service_name = 'signalfx-agent'
 
   if !$config['signalFxAccessToken'] {
     fail("The \$config parameter must contain a 'signalFxAccessToken'")
   }
 
+  if $::osfamily == 'windows' {
+    $agent_location = "C:\\Program Files\\SignalFx\\"
+    $split_config_file_path = $config_file_path.split("\\\\")
+    $config_parent_directory_path = $split_config_file_path[0, - 2].join("\\")
+
+    package { $service_name:
+      name            => 'signalfx-agent',
+      provider        => 'windows',
+      ensure          => 'installed',
+      source          => "${agent_location}\\SignalFxAgent\\bin\\signalfx-agent.exe",
+      install_options => [{ '-service' => '"install"' }, '-logEvents', { '-config' => $config_file_path }],
+    }
+  }
+  else {
+    $split_config_file_path = $config_file_path.split("/")
+    $config_parent_directory_path = $split_config_file_path[0, - 2].join("/")
+
+    unless $package_version == 'latest' {
```

Comment:
What about if `agent_version` is set to something and `package_version` is left at the default of `latest`?  Where does that come into play?

Reply:
Updated this so that `package_version` will take precedence if both are set.